lib/upgradable-multisig/multisig/utils.ak

```diff
@@ -0,0 +1,70 @@
+use aiken/list
+use aiken/transaction.{Input, Output}
+use aiken/transaction/credential.{Credential}
+use aiken/transaction/value.{PolicyId, Value}
+use upgradable_multisig/types.{AssetClass, MultisigDatum, PubKeyHash}
+
+pub fn get_input_by_token(
+  inputs: List<Input>,
+  policy_id: PolicyId,
+  token_name: ByteArray,
+) -> Option<Input> {
+  list.find(
+    inputs,
+    fn(input) {
+      value.quantity_of(input.output.value, policy_id, token_name) > 0
+    },
+  )
+}
+
+pub fn validate_multisig_datum(datum: MultisigDatum) -> Bool {
+  let signers_nr = list.length(datum.signers)
+
+  (signers_nr > 0)? && (datum.threshold > 0)? && (datum.spending_limit > 2_000_000)?
```

Comment:
```suggestion
  (signers_nr > 0)? && (signers_nr <= datum.threshold)? && (datum.threshold > 0)? && (datum.spending_limit > 2_000_000)?
```

Reply:
The signers have to always be more than the threshold so I can do !(signers_nr <= datum.threshold)?